crates/amaru-ledger/src/rules/block/header_size.rs

```diff
@@ -16,28 +16,23 @@ use amaru_kernel::protocol_parameters::ProtocolParameters;
 
 use super::{BlockValidation, InvalidBlockDetails};
 
-#[derive(Debug)]
-pub enum InvalidBlockHeader {
-    SizeTooBig { supplied: usize, max: usize },
-}
-
-#[allow(clippy::panic)]
 pub fn block_header_size_valid(
     header: &[u8],
     protocol_params: &ProtocolParameters,
-) -> BlockValidation {
+) -> BlockValidation<(), anyhow::Error> {
     let max_header_size = protocol_params
         .max_header_size
         .try_into()
-        .unwrap_or_else(|_| panic!("Failed to convert u32 to usize"));
+        .map(BlockValidation::Valid)
+        .unwrap_or_else(|e| BlockValidation::anyhow(e).context("failed to convert u32 to usize"))?;
```

Comment:
With a proper error type we wouldn't need to use `anyhow` here?

Reply:
Yes, but those are exceptions; and I don't think we really care to turn them into user facing errors. 

So it seems to be a perfect use case for anyhow?